<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-03-17" />

<title>Introduction</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Protocol for my lab-rotation with the Saez Rodriguez group</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Table of contents</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="MistyRStructuralAnalysisPipelineC2L.html">Structural analysis with MISTy - based on cell2location deconvolution</a>
    </li>
    <li>
      <a href="MistyRStructuralAnalysisPipelineDOT.html">Structural analysis with MISTy - based on DOT deconvolution</a>
    </li>
    <li>
      <a href="FunctionalPipelinePathwaySpecific.html">Functional analysis with MISTy - pathway specific genes</a>
    </li>
    <li>
      <a href="FunctionalPipelinePathwayActivityLigands.html">Functional analysis with MISTy - pathway activity and ligand expression</a>
    </li>
    <li>
      <a href="FunctionalAndStructuralPipeline.html">Learning functional and structural spatial relationships with MISTy</a>
    </li>
    <li>
      <a href="ReproduceSignaturePaper.html">MISTy representation based analysis of IMC breast cancer data</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/leotenshii/protocol">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Introduction</h1>
<h4 class="author">Leoni Zimmermann</h4>
<address class="author_afil">
Heidelberg University, Heidelberg,
Germany<br><h4 class="date">2024-03-17</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-03-17
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>ProtocolLabRotationSaezRodriguezGroup/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240306code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240306)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240306code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240306)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomleotenshiiProtocolLabRotationSaezRodriguezGrouptree5f6f819d0b2f3cf5a8d35a4d5e0bd6aeb27bdeaatargetblank5f6f819a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/leotenshii/ProtocolLabRotationSaezRodriguezGroup/tree/5f6f819d0b2f3cf5a8d35a4d5e0bd6aeb27bdeaa" target="_blank">5f6f819</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomleotenshiiProtocolLabRotationSaezRodriguezGrouptree5f6f819d0b2f3cf5a8d35a4d5e0bd6aeb27bdeaatargetblank5f6f819a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/leotenshii/ProtocolLabRotationSaezRodriguezGroup/tree/5f6f819d0b2f3cf5a8d35a4d5e0bd6aeb27bdeaa" target="_blank">5f6f819</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  data/10X_Visium_ACH005.tar.gz
    Untracked:  data/ACH005/
    Untracked:  data/bc_metadata.tsv
    Untracked:  data/hca_p14.rds
    Untracked:  data/imc_bc_optim_zoi.RDS
    Untracked:  data/omni_resource.csv
    Untracked:  omnipathr-log/
    Untracked:  result/
    Untracked:  results/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Introduction.Rmd</code>) and HTML
(<code>docs/Introduction.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/leotenshii/ProtocolLabRotationSaezRodriguezGroup/blob/5f6f819d0b2f3cf5a8d35a4d5e0bd6aeb27bdeaa/analysis/Introduction.Rmd" target="_blank">5f6f819</a>
</td>
<td>
leotenshii
</td>
<td>
2024-03-17
</td>
<td>
wflow_publish(c("analysis/_site.yml", "analysis/Introduction.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="general" class="section level2">
<h2>General</h2>
<p>Single-cell technologies have revolutionized the understanding of
cellular processes, by enabling the discovery of new gene regulation
mechanisms and protein expression dynamics. A significant limitation of
single-cell methods is that they take the cells out of the spatial
context, which is thus lost, and prevents unraveling the complex
interplay between cells in tissues. To address this challenge, spatial
omics methods have emerged enabling the analysis of different
biomolecules directly in tissue samples, conserving the spatial context.
These methods bridge the gap between single-cell resolution and spatial
information and provide invaluable insights into cellular interactions,
tissue structure, and disease mechanisms. In this context, spatial
transcriptomics and spatial proteomics have proven to be powerful tools
for analyzing the spatial organization of RNA and proteins in
tissues.</p>
</div>
<div id="spatial-transcriptomics" class="section level2">
<h2>Spatial transcriptomics</h2>
<p>Spatial transcriptomics allows for analyzing RNA-seq data at a
spatial level. This is possible through capturing both the spatial
context and transcriptional patterns at the same time. Yue et al. <span
class="citation">(<a
href="#ref-ReviewSpatialTranscriptomics"><em>1</em></a>)</span>
categorize the methods into four groups.</p>
<p>In situ hybridization-based technologies use labeled complementary
probes to bind and detect specific mRNA targets in tissue samples. The
second group includes in situ sequencing-based technologies. RNA
molecules are sequenced directly in the tissue sample. Various methods
can be used in this process, such as RNA reverse transcription, cDNA
cyclization, and rolling-circle amplification. Next generation
sequencing-based technologies form the next group. These technologies
capture RNA in tissue sections through RNA-capturing slides or DNA
barcoding. The last group is the spatial information reconstruction
technologies that combine RNA sequencing with computational methods like
image reconstruction or iterative algorithms.</p>
<p>Choosing the appropriate spatial transcriptomic technology involves
weighing factors such as gene throughput, sensitivity, resolution, and
feasibility against each other. Therefore, it is all the more important
to know the advantages and limitations of the methods to choose the
right one for an experiment.</p>
<div id="visium-10x" class="section level3">
<h3>Visium 10x</h3>
<p>Visium <span class="citation">(<a
href="#ref-Visium"><em>2</em></a>)</span> is a spatial transcriptomics
technology provided by 10x Genomics. It belongs to the group of Next
generation sequencing-based technologies working with RNA-capturing
slides. Tissue sections are placed on a special slide with capture
areas. Each of these capture areas contains 5000 barcoded spots. The
tissue sample is then stained and imaged routinely. Subsequently, the
tissue sample is fixed and permeabilized, allowing the capture of RNA on
the probes in the capture areas. Then, the RNA is sequenced, and a
sequencing library is constructed. This way, the transcripts can be
linked to their respective location on the tissue slide.</p>
<p>Visium can profile the whole transcriptome with a spatial resolution
of 55µm, corresponding to the size of a spot. Therefore, it can not
provide single-cell resolution since eukaryotic cells have an average
measure of 10µm. Further, its RNA capturing efficiency is relatively low
<span class="citation">(<a
href="#ref-VisiumAdvantagesDisadvantages"><em>3</em></a>)</span>. Still,
Visium is one of the most popular commercially available spatial
transcriptomics assays.</p>
</div>
</div>
<div id="spatial-proteomics" class="section level2">
<h2>Spatial Proteomics</h2>
<p>Spatial proteomics enables the analysis of protein localization
within tissues, providing insights into their spatial patterns.
Image-based methods visualize protein locations in the tissues through
specific targeting with antibodies <span class="citation">(<a
href="#ref-SpatialProteomics"><em>4</em></a>)</span>. The review groups
the methods in cyclic fluorescent and one-step mass-tag procedures. Both
are based on the same principle of antibody tagging with fluorophores,
DNA barcodes, enzymes, or metal tags and subsequent fluorescence imaging
or mass spectrometry.</p>
<p>Cyclic fluorescent approaches work by staining, imaging, and
extinguishing the signal, which is repeated for each protein to be
measured. Only one round of staining is needed when antibodies are
DNA-barcoded, making the assay faster. In contrast, one-step mass-tag
methods work without cycling but with one-time immunostaining and
detection. The applied antibodies are conjugated with ionizable metal
mass tags. Following the staining, mass spectrometry is used for data
acquisition. These methods have the advantage of being more robust than
cyclic fluorescent methods but also have the disadvantage of being more
expensive.</p>
<p>In comparison to spatial transcriptomics, these methods provide a
better estimation of cellular activity as RNA levels and protein levels
do not always correlate.</p>
<div id="image-mass-cytometry-imc" class="section level3">
<h3>Image Mass Cytometry (IMC)</h3>
<p>IMC <span class="citation">(<a href="#ref-IMC"><em>5</em></a>)</span>
belongs to the group of one-step mass-tag technologies. The tissue is
stained with antibodies tagged with a unique rare-earth-metal isotope.
The tissue is then ablated in a grid using a laser with a spot size of 1
µm. The ablated tissue is then transferred to a CyTOF mass cytometer.
The signal of the individual isotopes is extracted for all measured
markers, plotted according to the spot coordinates, and layered to
create a high-dimensional image. In the final step, a single-cell
segmentation is performed before the downstream data analysis.</p>
<p>IMC can image up to 40 markers simultaneously. It is estimated that
it will be possible to image over 100 markers in the future. IMC is also
faster than comparable methods such as Multiplexed Ion Beam Imaging
(MIBI) but at the expense of an increased spot size.</p>
</div>
</div>
<div id="misty" class="section level2">
<h2>MISTy</h2>
<p>As the number of spatial omics methods increases, so does the number
of applications for analyzing them. Examples are tools for neighborhood
analysis (e.g. Squidpy <span class="citation">(<a
href="#ref-Squidpy"><em>6</em></a>)</span>), finding spatial variable
genes (e.g. SpatialDE <span class="citation">(<a
href="#ref-SpatialDE"><em>7</em></a>)</span>), or spatial deconvolution
(e.g. Cell2location <span class="citation">(<a
href="#ref-C2L"><em>8</em></a>)</span>). Discovering interesting spatial
patterns through these methods raises questions about the interactions
that create them. One approach to answering this question involves the R
package MISTy (Multiview Intercellular SpaTial modeling framework) <span
class="citation">(<a href="#ref-MISTy"><em>9</em></a>)</span>.</p>
<p>MISTy uses an explainable machine learning algorithm to analyze
spatial omics data sets within and between spatial contexts called
views. The baseline of the model is the intraview that captures the
relationships within the spatial unit, be it a spot or a cell. Adding
further views is optional. Two built-in views that consider spatial
context are juxtaview and paraview. The juxtaview considers all marker
expressions in the close neighbors of a spatial unit, while the paraview
includes the marker expressions in the broader tissue weighted by their
distance to the spatial unit. The user can set the radius of these views
and add custom views. Then, for each view, the expression of a marker is
predicted based on the expression of all other markers in the spatial
context of the view. By default, this is modeled by a random forest, but
other algorithms are available. In the second step, the contribution of
each view in predicting the marker expression is estimated by ridge
regression of the beforehand obtained results. The output contains
performance, contribution, and importance estimations. The downstream
analysis can provide answers to the following questions:</p>
<ol style="list-style-type: decimal">
<li><p>To what extent can the analyzed marker expression from
surrounding tissues explain the marker expression of the spot? How does
it compare to the estimation of the intraview alone? There are three
different statistics to look at: R2 (the fraction of variance explained)
by the intraview and the multiview model and the resulting gain in R2
from the non-intraviews. This can be used to determine which marker
expressions are better explained when influences from outside the spot
are taken into account. Furthermore, it can be visualized which view has
how much influence on the improved prediction.</p></li>
<li><p>What are the specific relations that can explain the
contributions? To explain the contributions, the importance of each
marker in predicting all marker expressions is visualized separately for
each view. Interesting predictor-target pairs can then be visualized and
analyzed further with different tools.</p></li>
<li><p>What drives differences in the
performance/contribution/importance measurements? After training the
model, in the result space, these samples are represented by a vector
consisting of the sample signatures. There are three signatures:
performance, contribution, and importance. Based on the signatures,
causes for differences in performance metrics between the samples can be
analyzed. This is done by performing PCA on the signatures and finding
factors responsible for the clustering of the samples.</p></li>
</ol>
</div>
<div id="my-project" class="section level2">
<h2>My project</h2>
<p>The goal of my project was to add new vignettes for the MISTy package
since they are the most frequently accessed part of the MISTy website. I
have written a total of six vignettes that show how to answer the three
questions described above.</p>
<p>For five of the vignettes, I have used a Visium 10X dataset from
Kuppe et al. <span class="citation">(<a
href="#ref-VisiumHeart"><em>10</em></a>)</span>. They generated a
comprehensive map of cardiac remodeling following myocardial infarction
in humans. Multimodal data integration of single-cell gene expression,
chromatin accessibility, and spatial transcriptomic data was performed
to determine, for example, cell type composition. I selected patient
14’s (P14) data for the vignettes. Two of the vignettes are about
structural relationships, where I analyze the tissue slide based on the
cell type distribution. The other three are about functional
relationships, where I analyze the spatial patterns of the markers.
Selected markers are for example receptors, ligands, or pathway-specific
genes.</p>
<p>The sixth vignette demonstrates the sample signature analysis on an
IMC dataset <span class="citation">(<a
href="#ref-IMCBC"><em>11</em></a>)</span> of tumors of three different
grades, measuring 26 protein markers. In the vignette, I show how the
samples can be distinguished by their signatures and that this grouping
is similar to that by grade or clinical type.</p>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<div id="refs" class="references csl-bib-body">
<div id="ref-ReviewSpatialTranscriptomics" class="csl-entry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">L.
Yue <em>et al.</em>, <a
href="https://doi.org/10.1016/j.csbj.2023.01.016">A guidebook of spatial
transcriptomic technologies, data resources and analysis approaches</a>.
<em>Computational and Structural Biotechnology Journal</em>.
<strong>21</strong>, 940–955 (2023).</div>
</div>
<div id="ref-Visium" class="csl-entry">
<div class="csl-left-margin">2. </div><div
class="csl-right-inline">Visium spatial gene expression (14/03/2024)
(available at <a
href="https://www.10xgenomics.com/products/spatial-gene-expression">https://www.10xgenomics.com/products/spatial-gene-expression</a>).</div>
</div>
<div id="ref-VisiumAdvantagesDisadvantages" class="csl-entry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Y.
Wang <em>et al.</em>, <a
href="https://doi.org/10.1016/j.ygeno.2023.110671">Spatial
transcriptomics: Technologies, applications and experimental
considerations</a>. <em>Genomics</em>. <strong>115</strong>, 110671
(2023).</div>
</div>
<div id="ref-SpatialProteomics" class="csl-entry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">J.
R. Moffitt, E. Lundberg, H. Heyn, <a
href="https://doi.org/10.1038/s41576-022-00515-3">The emerging landscape
of spatial profiling technologies</a>. <em>Nat Rev Genet</em>.
<strong>23</strong>, 741–759 (2022).</div>
</div>
<div id="ref-IMC" class="csl-entry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">C.
Giesen <em>et al.</em>, <a
href=" https://doi.org/10.1038/nmeth.2869">Highly multiplexed imaging of
tumor tissues with subcellular resolution by mass cytometry</a>. <em>Nat
Methods</em>. <strong>11</strong>, 417–22 (2014).</div>
</div>
<div id="ref-Squidpy" class="csl-entry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline">G.
Palla <em>et al.</em>, <a
href="https://doi.org/10.1038/s41592-021-01358-2">Squidpy: A scalable
framework for spatial omics analysis</a>. <em>Nat Methods</em>.
<strong>19</strong>, 171–178 (2022).</div>
</div>
<div id="ref-SpatialDE" class="csl-entry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline">V.
Svensson, S. A. Teichmann, O. Stegle, <a
href="https://doi.org/10.1038/nmeth.4636">SpatialDE: Identification of
spatially variable genes</a>. <em>Nat Methods</em>. <strong>15</strong>,
343–346 (2018).</div>
</div>
<div id="ref-C2L" class="csl-entry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline">V.
Kleshchevnikov <em>et al.</em>, <a
href="https://doi.org/10.1038/s41587-021-01139-4">Cell2location maps
fine-grained cell types in spatial transcriptomics</a>. <em>Nat
Biotechnol</em>. <strong>40</strong>, 661–671 (2022).</div>
</div>
<div id="ref-MISTy" class="csl-entry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline">J.
Tanevski, R. O. R. Flores, A. Gabor, D. Schapiro, J. Saez-Rodriguez, <a
href="https://doi.org/10.1186/s13059-022-02663-5">Explainable multiview
framework for dissecting spatial relationships from highly multiplexed
data</a>. <em>Genome Biol</em>. <strong>23</strong>, 97 (2022).</div>
</div>
<div id="ref-VisiumHeart" class="csl-entry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline">C.
Kuppe <em>et al.</em>, <a
href="https://doi.org/10.1038/s41586-022-05060-x">Spatial multi-omic map
of human myocardial infarction</a>. <em>Nature</em>.
<strong>608</strong>, 766–777 (2022).</div>
</div>
<div id="ref-IMCBC" class="csl-entry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline">S.
D <em>et al.</em>, <a
href="https://doi.org/10.1038/nmeth.4391">histoCAT: Analysis of cell
phenotypes and interactions in multiplex image cytometry data.</a>
<em>Nat Methods</em>. <strong>14</strong>, 873–876 (2017).</div>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=German_Germany.utf8  LC_CTYPE=German_Germany.utf8   
[3] LC_MONETARY=German_Germany.utf8 LC_NUMERIC=C                   
[5] LC_TIME=German_Germany.utf8    

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.1

loaded via a namespace (and not attached):
 [1] vctrs_0.6.5       httr_1.4.7        cli_3.6.2         knitr_1.45       
 [5] rlang_1.1.2       xfun_0.41         stringi_1.8.3     processx_3.8.3   
 [9] promises_1.2.1    jsonlite_1.8.8    glue_1.6.2        rprojroot_2.0.4  
[13] git2r_0.33.0      htmltools_0.5.7   httpuv_1.6.13     ps_1.7.5         
[17] sass_0.4.8        fansi_1.0.6       rmarkdown_2.25    jquerylib_0.1.4  
[21] tibble_3.2.1      evaluate_0.23     fastmap_1.1.1     yaml_2.3.8       
[25] lifecycle_1.0.4   whisker_0.4.1     stringr_1.5.1     compiler_4.3.2   
[29] fs_1.6.3          pkgconfig_2.0.3   Rcpp_1.0.11       rstudioapi_0.15.0
[33] later_1.3.2       digest_0.6.33     R6_2.5.1          utf8_1.2.4       
[37] pillar_1.9.0      callr_3.7.3       magrittr_2.0.3    bslib_0.6.1      
[41] tools_4.3.2       cachem_1.0.8      getPass_0.2-4    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
