<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-03-11" />

<title>Structural analysis with MISTy - based on cell2location deconvolution</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Protocol for my lab-rotation with the Saez Rodriguez group</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Table of contents</a>
</li>
<li>
  <a href="MistyRStructuralAnalysisPipelineC2L.Rmd">Structural analysis with MISTy - based on cell2location deconvolution</a>
</li>
<li>
  <a href="MistyRStructuralAnalysisPipelineDOT.Rmd">Structural analysis with MISTy - based on DOT deconvolution</a>
</li>
<li>
  <a href="FunctionalPipelinePathwaySpecific.Rmd">Functional analysis with MISTy - pathway specific genes</a>
</li>
<li>
  <a href="FunctionalPipelinePathwayActivityLigands.Rmd">Functional analysis with MISTy - pathway activity and ligand expression</a>
</li>
<li>
  <a href="FunctionalAndStructuralPipeline.Rmd">Learning functional and structural spatial relationships with MISTy</a>
</li>
<li>
  <a href="ReproduceSignaturePaper.Rmd">MISTy representation based analysis of IMC breast cancer data</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/leotenshii/protocol">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Structural analysis with MISTy - based on
cell2location deconvolution</h1>
<h4 class="author">Leoni Zimmermann</h4>
<address class="author_afil">
Heidelberg University, Heidelberg, Germany<br><h4 class="author">Jovan
Tanevski</h4>
<address class="author_afil">
Heidelberg University and Heidelberg University Hospital, Heidelberg,
GermanyJožef Stefan Institute, Ljubljana,
Slovenia<br><a class="author_email" href="mailto:#"><a
href="mailto:jovan.tanevski@uni-heidelberg.de"
class="email">jovan.tanevski@uni-heidelberg.de</a></a>
</address>
<h4 class="date">2024-03-11</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-03-11
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>ProtocolLabRotationSaezRodriguezGroup/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240306code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240306)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240306code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240306)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomleotenshiiProtocolLabRotationSaezRodriguezGrouptree38a9f93e85199971b1674f15a9a7b528ec582d8btargetblank38a9f93a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/leotenshii/ProtocolLabRotationSaezRodriguezGroup/tree/38a9f93e85199971b1674f15a9a7b528ec582d8b" target="_blank">38a9f93</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomleotenshiiProtocolLabRotationSaezRodriguezGrouptree38a9f93e85199971b1674f15a9a7b528ec582d8btargetblank38a9f93a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/leotenshii/ProtocolLabRotationSaezRodriguezGroup/tree/38a9f93e85199971b1674f15a9a7b528ec582d8b" target="_blank">38a9f93</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  10X_Visium_ACH005.tar.gz
    Untracked:  ACH005/
    Untracked:  data/10X_Visium_ACH005.tar.gz
    Untracked:  omni_resource.csv
    Untracked:  omnipathr-log/
    Untracked:  result/
    Untracked:  results/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/MistyRStructuralAnalysisPipelineC2L.Rmd</code>) and HTML
(<code>docs/MistyRStructuralAnalysisPipelineC2L.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/leotenshii/ProtocolLabRotationSaezRodriguezGroup/blob/38a9f93e85199971b1674f15a9a7b528ec582d8b/analysis/MistyRStructuralAnalysisPipelineC2L.Rmd" target="_blank">38a9f93</a>
</td>
<td>
leotenshii
</td>
<td>
2024-03-11
</td>
<td>
Revert "changed paths to data/…"
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/leotenshii/ProtocolLabRotationSaezRodriguezGroup/blob/5e9bf15197b10ae67cdd54cd23cb270805288e35/analysis/MistyRStructuralAnalysisPipelineC2L.Rmd" target="_blank">5e9bf15</a>
</td>
<td>
leotenshii
</td>
<td>
2024-03-11
</td>
<td>
changed paths to data/…
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/leotenshii/ProtocolLabRotationSaezRodriguezGroup/blob/5ec41b8847e25c78b65644368fcf59c3f6654c4f/analysis/MistyRStructuralAnalysisPipelineC2L.Rmd" target="_blank">5ec41b8</a>
</td>
<td>
leotenshii
</td>
<td>
2024-03-11
</td>
<td>
upload of vignettes
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>MISTy is designed to analyze spatial omics datasets within and
between distinct spatial contexts referred to as views. This analysis
can focus solely on structural information. Spatial transcriptomic
methods such as 10x Visium capture information from areas containing
multiple cells. Then, deconvolution is applied to relate the measured
data of the spots back to individual cells. A commonly used tool for
this deconvolution step is cell2location.</p>
<p>This vignette presents a workflow for the analysis of structural
data, guiding users through the application of <code>mistyR</code> to
the results of <code>cell2location</code> deconvolution.</p>
<p>Load the necessary packages:</p>
<pre class="r"><code># MISTy 
library(mistyR) 
library(future) 

#Seurat 
library(Seurat)
library(SeuratObject)

# Data manipulation 
library(tidyverse) </code></pre>
<pre><code>Warning: Paket &#39;ggplot2&#39; wurde unter R Version 4.3.3 erstellt</code></pre>
<pre><code>Warning: Paket &#39;readr&#39; wurde unter R Version 4.3.3 erstellt</code></pre>
<pre class="r"><code># Distances
library(distances)</code></pre>
</div>
<div id="get-and-load-the-data" class="section level2">
<h2>Get and load the data</h2>
<p>For this showcase, we use a 10X Visium spatial slide from <a
href="https://doi.org/10.1038/s41586-022-05060-x">Kuppe et al.,
2022</a>, where they created a spatial multi-omic map of human
myocardial infarction. The example data comes from a human heart in a
later state after myocardial infarction that was used in the study. The
Seurat object contains, among other things, the coordinates of the spots
on the slides and their cellular composition estimated by cell2location.
First, we have to download and extract the file:</p>
<pre class="r"><code>download.file(&quot;https://zenodo.org/records/6580069/files/10X_Visium_ACH005.tar.gz?download=1&quot;,
    destfile = &quot;10X_Visium_ACH005.tar.gz&quot;, method = &quot;curl&quot;)
untar(&quot;10X_Visium_ACH005.tar.gz&quot;)</code></pre>
<p>The next step is to load the data and extract the cell composition
and location of the spots. The rows are shifted, which means that the
real distances between two spots are not always the same. It is
therefore advantageous to use the pixel coordinates instead of row and
column numbers, as the distances between these are represented
accurately.</p>
<pre class="r"><code># Load file into R
seurat_vs &lt;- readRDS(&quot;ACH005/ACH005.rds&quot;)

# Extract the cell composition
composition &lt;- as_tibble(t(seurat_vs[[&quot;c2l_props&quot;]]$data)) 

# Extract the location data
geometry &lt;- GetTissueCoordinates(seurat_vs, cols = c(&quot;imagerow&quot;, &quot;imagecol&quot;), scale = NULL)</code></pre>
</div>
<div id="visualize-cell-proportion-in-spots" class="section level2">
<h2>Visualize cell proportion in spots</h2>
<p>First, we visually explore the slide itself and then through a
graphical representation of cell-type proportions at each spot. When
adding <code>interactive = TRUE</code> to
<code>SpatialFeaturePlot</code> we can cycle through the proportions of
the different celltypes.</p>
<pre class="r"><code># Tissue Slide
SpatialPlot(seurat_vs, keep.scale = NULL, alpha = 0) </code></pre>
<p><img src="figure/MistyRStructuralAnalysisPipelineC2L.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Cell type proportions
DefaultAssay(seurat_vs) &lt;- &quot;c2l_props&quot;
SpatialFeaturePlot(seurat_vs, keep.scale = NULL, features = &quot;CM&quot;) </code></pre>
<p><img src="figure/MistyRStructuralAnalysisPipelineC2L.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Based on the plots, we can observe that some cell types are found
more frequently than others. Additionally, we can identify patterns in
the distribution of cells, with some being widespread across the entire
slide while others are concentrated in specific areas. Furthermore,
there are cell types that share a similar distribution.</p>
</div>
<div id="misty-views" class="section level2">
<h2>MISTy views</h2>
<p>First, we need to define an intraview that captures the cell type
proportions within a spot. To capture the distribution of cell type
proportions in the surrounding tissue, we add a paraview. For this
vignette, the radius we choose is the distance to the nearest neighbor
plus the standard deviation. We calculate the weights of each spot with
<code>family = gaussian</code>. Then we run MISTy and collect the
results.</p>
<pre class="r"><code># Calculating the radius
geom_dist &lt;- as.matrix(distances(geometry))  
dist_nn &lt;- apply(geom_dist, 1, function(x) (sort(x)[2]))
paraview_radius &lt;- ceiling(mean(dist_nn+ sd(dist_nn)))

# Create views
heart_views &lt;- create_initial_view(composition) %&gt;%
  add_paraview(geometry, l= paraview_radius, family = &quot;gaussian&quot;)

# Run misty and collect results
run_misty(heart_views, &quot;result/vignette_structural_pipeline&quot;)</code></pre>
<pre><code>[1] &quot;F:/LabRotationSaez/ProtocolLabRotationSaezRodriguezGroup/result/vignette_structural_pipeline&quot;</code></pre>
<pre class="r"><code>misty_results &lt;- collect_results(&quot;result/vignette_structural_pipeline&quot;)</code></pre>
</div>
<div id="downstream-analysis" class="section level2">
<h2>Downstream Analysis</h2>
<p>With the collected results, we can now answer following
questions:</p>
<div
id="to-what-extent-can-the-occurring-cell-types-of-the-surrounding-tissue-explain-the-cell-type-composition-of-the-spot-compared-to-the-intraview"
class="section level4">
<h4>1. To what extent can the occurring cell types of the surrounding
tissue explain the cell type composition of the spot compared to the
intraview?</h4>
<p>Here we can look at two different statistics: <code>multi.R2</code>
shows the total variance explained by the multiview model.
<code>gain.R2</code> shows the increase in explainable variance from to
the paraview.</p>
<pre class="r"><code>misty_results %&gt;%
  plot_improvement_stats(&quot;multi.R2&quot;) %&gt;% 
  plot_improvement_stats(&quot;gain.R2&quot;)</code></pre>
<pre><code>Warning: Removed 11 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
<p><img src="figure/MistyRStructuralAnalysisPipelineC2L.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 11 rows containing missing values or values outside the scale range
(`geom_segment()`).</code></pre>
<p><img src="figure/MistyRStructuralAnalysisPipelineC2L.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>The paraview particularly increases the explained variance for
adipocytes and mast cells. In general, the significant gain in
R<sup>2</sup> can be interpreted as the following:</p>
<p>“We can better explain the expression of marker X when we consider
additional views other than the intrinsic view.”</p>
</div>
<div
id="what-are-the-specific-relations-that-can-explain-the-contributions"
class="section level4">
<h4>2. What are the specific relations that can explain the
contributions?</h4>
<p>To explain the contributions, we can visualize the importance of each
cell type in predicting the cell type distribution for each view
separately.</p>
<p>First, for the intrinsic view:</p>
<pre class="r"><code>misty_results %&gt;% plot_interaction_heatmap(view = &quot;intra&quot;, clean = TRUE)</code></pre>
<p><img src="figure/MistyRStructuralAnalysisPipelineC2L.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>We can observe that cardiomyocytes are a significant predictor for
several cell types when in the same spot. To identify the target with
the best prediction by cardiomyocytes, we can view the importance values
as follows:</p>
<pre class="r"><code>misty_results$importances.aggregated %&gt;%
  filter(view == &quot;intra&quot;, Predictor == &quot;CM&quot;) %&gt;%
  arrange(-Importance)</code></pre>
<pre><code># A tibble: 11 × 5
   view  Predictor Target   Importance nsamples
   &lt;chr&gt; &lt;chr&gt;     &lt;chr&gt;         &lt;dbl&gt;    &lt;int&gt;
 1 intra CM        Fib          2.74          1
 2 intra CM        vSMCs        2.43          1
 3 intra CM        prolif       2.23          1
 4 intra CM        Myeloid      2.21          1
 5 intra CM        Endo         2.15          1
 6 intra CM        Mast         2.02          1
 7 intra CM        PC           1.90          1
 8 intra CM        Lymphoid     0.844         1
 9 intra CM        Adipo       -0.0416        1
10 intra CM        Neuronal    -0.0865        1
11 intra CM        CM          NA             1</code></pre>
<p>Let’s take a look at the spatial distribution of these cell types in
the tissue slide:</p>
<pre class="r"><code>SpatialFeaturePlot(seurat_vs, keep.scale = NULL, features = c(&quot;Fib&quot;,&quot;CM&quot;), image.alpha = 0)</code></pre>
<p><img src="figure/MistyRStructuralAnalysisPipelineC2L.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>We can observe that areas with high proportions of cardiomyocytes
have low proportions of fibroblasts and vice versa.</p>
<p>Now we repeat this analysis with the paraview. With <code>trim</code>
we display only targets with a value above 1.75 for
<code>gain.R2</code>. To set an importance threshold we apply
<code>cutoff</code>.</p>
<pre class="r"><code>misty_results %&gt;% plot_interaction_heatmap(view = &quot;para.126&quot;, clean = TRUE, 
                                           trim = 1.75, trim.measure = &quot;gain.R2&quot;,
                                           cutoff = 0.5) </code></pre>
<p><img src="figure/MistyRStructuralAnalysisPipelineC2L.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Here, we select the target adipocytes, as we know from previous
analysis that the paraview contributes a large part to explaining its
distribution. The best predictor for adipocytes are proliferating
cells:</p>
<pre class="r"><code>SpatialFeaturePlot(seurat_vs, keep.scale = NULL, features = c(&quot;prolif&quot;,&quot;Adipo&quot;), image.alpha = 0)</code></pre>
<p><img src="figure/MistyRStructuralAnalysisPipelineC2L.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The plots show us that, in some places, the localization of the two
cell types overlap.</p>
</div>
</div>
<div id="see-also" class="section level2">
<h2>See also</h2>
<p><code>browseVignettes("mistyR")</code></p>
</div>
<div id="session-info" class="section level2">
<h2>Session Info</h2>
<p>Here is the output of <code>sessionInfo()</code> at the point when
this document was compiled.</p>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=German_Germany.utf8  LC_CTYPE=German_Germany.utf8   
[3] LC_MONETARY=German_Germany.utf8 LC_NUMERIC=C                   
[5] LC_TIME=German_Germany.utf8    

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] distances_0.1.10   lubridate_1.9.3    forcats_1.0.0      stringr_1.5.1     
 [5] dplyr_1.1.4        purrr_1.0.2        readr_2.1.5        tidyr_1.3.0       
 [9] tibble_3.2.1       ggplot2_3.5.0      tidyverse_2.0.0    Seurat_5.0.1      
[13] SeuratObject_5.0.1 sp_2.1-2           future_1.33.1      mistyR_1.10.0     
[17] workflowr_1.7.1   

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.21       splines_4.3.2          later_1.3.2           
  [4] filelock_1.0.3         R.oo_1.26.0            polyclip_1.10-6       
  [7] hardhat_1.3.1          pROC_1.18.5            rpart_4.1.23          
 [10] fastDummies_1.7.3      lifecycle_1.0.4        rprojroot_2.0.4       
 [13] vroom_1.6.5            globals_0.16.3         processx_3.8.3        
 [16] lattice_0.22-5         MASS_7.3-60            magrittr_2.0.3        
 [19] plotly_4.10.3          sass_0.4.8             rmarkdown_2.25        
 [22] jquerylib_0.1.4        yaml_2.3.8             rlist_0.4.6.2         
 [25] httpuv_1.6.13          sctransform_0.4.1      spam_2.10-0           
 [28] spatstat.sparse_3.0-3  reticulate_1.34.0      cowplot_1.1.2         
 [31] pbapply_1.7-2          RColorBrewer_1.1-3     abind_1.4-5           
 [34] Rtsne_0.17             R.utils_2.12.3         nnet_7.3-19           
 [37] ipred_0.9-14           git2r_0.33.0           lava_1.8.0            
 [40] ggrepel_0.9.4          irlba_2.3.5.1          listenv_0.9.1         
 [43] spatstat.utils_3.0-4   goftest_1.2-3          RSpectra_0.16-1       
 [46] spatstat.random_3.2-2  fitdistrplus_1.1-11    parallelly_1.37.1     
 [49] leiden_0.4.3.1         codetools_0.2-19       tidyselect_1.2.0      
 [52] farver_2.1.1           stats4_4.3.2           matrixStats_1.2.0     
 [55] spatstat.explore_3.2-5 jsonlite_1.8.8         caret_6.0-94          
 [58] ellipsis_0.3.2         progressr_0.14.0       ggridges_0.5.5        
 [61] survival_3.5-7         iterators_1.0.14       foreach_1.5.2         
 [64] tools_4.3.2            ica_1.0-3              Rcpp_1.0.11           
 [67] glue_1.6.2             prodlim_2023.08.28     gridExtra_2.3         
 [70] xfun_0.41              ranger_0.16.0          withr_3.0.0           
 [73] fastmap_1.1.1          fansi_1.0.6            callr_3.7.3           
 [76] digest_0.6.33          timechange_0.3.0       R6_2.5.1              
 [79] mime_0.12              colorspace_2.1-0       scattermore_1.2       
 [82] tensor_1.5             spatstat.data_3.0-3    R.methodsS3_1.8.2     
 [85] utf8_1.2.4             generics_0.1.3         data.table_1.15.2     
 [88] recipes_1.0.10         class_7.3-22           httr_1.4.7            
 [91] ridge_3.3              htmlwidgets_1.6.4      whisker_0.4.1         
 [94] ModelMetrics_1.2.2.2   uwot_0.1.16            pkgconfig_2.0.3       
 [97] gtable_0.3.4           timeDate_4032.109      lmtest_0.9-40         
[100] furrr_0.3.1            htmltools_0.5.7        dotCall64_1.1-1       
[103] scales_1.3.0           png_0.1-8              gower_1.0.1           
[106] knitr_1.45             rstudioapi_0.15.0      tzdb_0.4.0            
[109] reshape2_1.4.4         nlme_3.1-164           cachem_1.0.8          
[112] zoo_1.8-12             KernSmooth_2.23-22     parallel_4.3.2        
[115] miniUI_0.1.1.1         pillar_1.9.0           grid_4.3.2            
[118] vctrs_0.6.5            RANN_2.6.1             promises_1.2.1        
[121] xtable_1.8-4           cluster_2.1.6          archive_1.1.7         
[124] evaluate_0.23          cli_3.6.2              compiler_4.3.2        
[127] crayon_1.5.2           rlang_1.1.2            future.apply_1.11.1   
[130] labeling_0.4.3         ps_1.7.5               getPass_0.2-4         
[133] plyr_1.8.9             fs_1.6.3               stringi_1.8.3         
[136] viridisLite_0.4.2      deldir_2.0-4           assertthat_0.2.1      
[139] munsell_0.5.0          lazyeval_0.2.2         spatstat.geom_3.2-7   
[142] Matrix_1.6-4           RcppHNSW_0.5.0         hms_1.1.3             
[145] patchwork_1.2.0.9000   bit64_4.0.5            shiny_1.8.0           
[148] highr_0.10             ROCR_1.0-11            igraph_1.6.0          
[151] bslib_0.6.1            bit_4.0.5             </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=German_Germany.utf8  LC_CTYPE=German_Germany.utf8   
[3] LC_MONETARY=German_Germany.utf8 LC_NUMERIC=C                   
[5] LC_TIME=German_Germany.utf8    

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] distances_0.1.10   lubridate_1.9.3    forcats_1.0.0      stringr_1.5.1     
 [5] dplyr_1.1.4        purrr_1.0.2        readr_2.1.5        tidyr_1.3.0       
 [9] tibble_3.2.1       ggplot2_3.5.0      tidyverse_2.0.0    Seurat_5.0.1      
[13] SeuratObject_5.0.1 sp_2.1-2           future_1.33.1      mistyR_1.10.0     
[17] workflowr_1.7.1   

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.21       splines_4.3.2          later_1.3.2           
  [4] filelock_1.0.3         R.oo_1.26.0            polyclip_1.10-6       
  [7] hardhat_1.3.1          pROC_1.18.5            rpart_4.1.23          
 [10] fastDummies_1.7.3      lifecycle_1.0.4        rprojroot_2.0.4       
 [13] vroom_1.6.5            globals_0.16.3         processx_3.8.3        
 [16] lattice_0.22-5         MASS_7.3-60            magrittr_2.0.3        
 [19] plotly_4.10.3          sass_0.4.8             rmarkdown_2.25        
 [22] jquerylib_0.1.4        yaml_2.3.8             rlist_0.4.6.2         
 [25] httpuv_1.6.13          sctransform_0.4.1      spam_2.10-0           
 [28] spatstat.sparse_3.0-3  reticulate_1.34.0      cowplot_1.1.2         
 [31] pbapply_1.7-2          RColorBrewer_1.1-3     abind_1.4-5           
 [34] Rtsne_0.17             R.utils_2.12.3         nnet_7.3-19           
 [37] ipred_0.9-14           git2r_0.33.0           lava_1.8.0            
 [40] ggrepel_0.9.4          irlba_2.3.5.1          listenv_0.9.1         
 [43] spatstat.utils_3.0-4   goftest_1.2-3          RSpectra_0.16-1       
 [46] spatstat.random_3.2-2  fitdistrplus_1.1-11    parallelly_1.37.1     
 [49] leiden_0.4.3.1         codetools_0.2-19       tidyselect_1.2.0      
 [52] farver_2.1.1           stats4_4.3.2           matrixStats_1.2.0     
 [55] spatstat.explore_3.2-5 jsonlite_1.8.8         caret_6.0-94          
 [58] ellipsis_0.3.2         progressr_0.14.0       ggridges_0.5.5        
 [61] survival_3.5-7         iterators_1.0.14       foreach_1.5.2         
 [64] tools_4.3.2            ica_1.0-3              Rcpp_1.0.11           
 [67] glue_1.6.2             prodlim_2023.08.28     gridExtra_2.3         
 [70] xfun_0.41              ranger_0.16.0          withr_3.0.0           
 [73] fastmap_1.1.1          fansi_1.0.6            callr_3.7.3           
 [76] digest_0.6.33          timechange_0.3.0       R6_2.5.1              
 [79] mime_0.12              colorspace_2.1-0       scattermore_1.2       
 [82] tensor_1.5             spatstat.data_3.0-3    R.methodsS3_1.8.2     
 [85] utf8_1.2.4             generics_0.1.3         data.table_1.15.2     
 [88] recipes_1.0.10         class_7.3-22           httr_1.4.7            
 [91] ridge_3.3              htmlwidgets_1.6.4      whisker_0.4.1         
 [94] ModelMetrics_1.2.2.2   uwot_0.1.16            pkgconfig_2.0.3       
 [97] gtable_0.3.4           timeDate_4032.109      lmtest_0.9-40         
[100] furrr_0.3.1            htmltools_0.5.7        dotCall64_1.1-1       
[103] scales_1.3.0           png_0.1-8              gower_1.0.1           
[106] knitr_1.45             rstudioapi_0.15.0      tzdb_0.4.0            
[109] reshape2_1.4.4         nlme_3.1-164           cachem_1.0.8          
[112] zoo_1.8-12             KernSmooth_2.23-22     parallel_4.3.2        
[115] miniUI_0.1.1.1         pillar_1.9.0           grid_4.3.2            
[118] vctrs_0.6.5            RANN_2.6.1             promises_1.2.1        
[121] xtable_1.8-4           cluster_2.1.6          archive_1.1.7         
[124] evaluate_0.23          cli_3.6.2              compiler_4.3.2        
[127] crayon_1.5.2           rlang_1.1.2            future.apply_1.11.1   
[130] labeling_0.4.3         ps_1.7.5               getPass_0.2-4         
[133] plyr_1.8.9             fs_1.6.3               stringi_1.8.3         
[136] viridisLite_0.4.2      deldir_2.0-4           assertthat_0.2.1      
[139] munsell_0.5.0          lazyeval_0.2.2         spatstat.geom_3.2-7   
[142] Matrix_1.6-4           RcppHNSW_0.5.0         hms_1.1.3             
[145] patchwork_1.2.0.9000   bit64_4.0.5            shiny_1.8.0           
[148] highr_0.10             ROCR_1.0-11            igraph_1.6.0          
[151] bslib_0.6.1            bit_4.0.5             </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
